packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -229,13 +261,62 @@ export default createRule<Options, MessageIds>({
             }
           })();
 
+          const typeAnnotation = (() => {
+            if (esNode.type !== AST_NODE_TYPES.PropertyDefinition) {
+              return null;
+            }
+
+            if (esNode.typeAnnotation || !esNode.value) {
+              return null;
+            }
+
+            if (nameNode.type !== AST_NODE_TYPES.Identifier) {
+              return null;
+            }
```

Comment:
To my understanding, this can realistically only be an identifier. This means some cases that should have a type annotation added won't receive it, for example:

```ts
class TestOverlappingClassVariable {
  private 'overlappingClassVariable' = 7;

  constructor() {
    this.overlappingClassVariable = 10;
  }
}
```

This seems like an existing bug that's not necessarily related to this addition. I've opened #10655 as a separate issue to address this.